components/camera/align/join.go

```diff
@@ -90,20 +91,44 @@ type joinColorDepth struct {
 // newJoinColorDepth creates a gostream.VideoSource that aligned color and depth channels.
 func newJoinColorDepth(ctx context.Context, color, depth camera.VideoSource, conf *joinConfig, logger golog.Logger,
 ) (camera.VideoSource, error) {
+	imgType := camera.ImageType(conf.ImageType)
+	// get intrinsic parameters from config, or from the underlying camera
+	var camParams *transform.PinholeCameraIntrinsics
 	if conf.CameraParameters == nil {
-		return nil, errors.Wrap(transform.ErrNoIntrinsics, "intrinsic_parameters field in attributes cannot be empty")
+		if imgType == camera.DepthStream {
+			props, err := depth.Properties(ctx)
+			if err == nil {
+				camParams = props.IntrinsicParams
+			}
+		} else {
+			props, err := color.Properties(ctx)
+			if err == nil {
+				camParams = props.IntrinsicParams
+			}
+		}
+	} else {
+		camParams = conf.CameraParameters
+	}
```

Comment:
This is to grab the intrinsics from the underlying camera if the intrinsics are not provided in the config attributes.